.github/workflows/ci.yaml

```diff
@@ -1,95 +1,85 @@
 name: ci
 on: [push, pull_request]
 jobs:
-  test-27:
+  test:
     runs-on: ${{ matrix.platform }}
     strategy:
       matrix:
-        platform: [ ubuntu-latest ]
+        platform: [ubuntu-latest]
+        python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
+
     env:
-      NOSE_SHOW_SKIPPED: 1
-      PY_COLOR: 1
+      PY_COLORS: 1
+
     steps:
       - uses: actions/checkout@v2
-      - name: Set up Python 2.7
+
+      - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
-          python-version: 2.7
-      - uses: actions/cache@v1
-        if: startsWith(runner.os, 'Linux')
-        with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
-          restore-keys: |
-            ${{ runner.os }}-pip-
-      - uses: actions/cache@v1
-        if: startsWith(runner.os, 'Windows')
-        with:
-          path: ~\AppData\Local\pip\Cache
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
-          restore-keys: |
-            ${{ runner.os }}-pip-
+          python-version: ${{ matrix.python-version }}
+
       - name: Install base dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install tox sphinx
+          python -m pip install tox sphinx
```

Comment:
I don't understand why we have to install sphinx here, rather then it getting installed as needed in tox?

Reply:
I believe (although am not 100% certain) that this is because Sphinx is used in the sdist building step, i.e., it's a build-time dependency rather than a test-time dependency.